source/_posts/Docker-Swarm.md

```diff
@@ -0,0 +1,346 @@
+---
+title: Docker Swarm, Traefik, Let's Encrypt en Portainer
+date: 2019-11-03 12:00:00
+tags:
+- Docker swarm
+- Let's Encrypt
+- Traefik
+- Portainer
+---
+
+Case omschrijving 
+---
+Niet lang geleden heb ik besloten om mij eens goed te gaan verdiepen in Docker. Je hoort het overal en het wordt steeds populairder en ik hoor steeds vaker dat het een goed alternatief is voor de huidige manier van werken namelijk Faas en Paas in een public cloud hierdoor ben ik nieuwsgierig geworden en wil ik eens kijken wat hier nu de voordelen van zijn. Ik ben begonnen om mij te certificeren, ik heb een cursus gevolgd en mijn examen met succes behaald. Ik vind het altijd belangrijk om te weten hoe iets in elkaar zit en waarom het zo in elkaar zit. Mijn idee is om een of meerdere productie applicaties op een Docker Swarm cluster te gaan draaien. In deze blog ga ik laten zien hoe ik een goede infrastructuur bouw om de applicaties te hosten.
+
+De omgeving moest aan de volgende voorwaarden voldoen:
+
+- Er moeten meerdere web applicaties op kunnen draaien welke op poort 80 en 443 benaderd kunnen worden.
+- Alle web applicaties moeten draaien op SSL, dit moet mij zo min mogelijk werk kosten.
+- De oplossing moet schaalbaar zijn en bestand zijn tegen het uitvallen en/of offline gaan van servers.
+- Ik wil de oplossing kunnen hosten bij een Cloud provider, Hosting provider of lokaal om te kunnen testen.
+- Het beheren van de containers moet gemakkelijk en overzichtelijk zijn.
+- Aangezien ik een echte Hollander ben moeten de kosten ook een beetje beperkt blijven.
+
+De oplossing welke ik heb gemaakt is een combinatie van Docker Swarm, Traefik, Let's Encrypt en Portainer.
+
+Wat is Docker Swarm
+---
+Docker Swarm is een tool waarmee je Docker-containers kunt beheren en schalen. Als je Docker installeert dan krijg je daar meteen Swarm bij. Docker Swarm is een standaard product van Docker wat bij elke installatie van Docker wordt meegeleverd. Met Docker Swarm kun je een cluster bouwen van verschillende virtuele machines deze worden hierna nodes genoemd. Hierop kunnen de Docker-containers worden gedeployed als een stack\*\* of een service\*. Als je een cluster wilt hebben wat een hoge beschikbaarheid heeft wordt aangeraden om minstens 3 manager nodes te hebben. Docker Swarm maakt namelijk gebruik van het Raft Consensus Algoritme, 1 manager is de leider van de swarm en de status van de manager wordt gesynchroniseerd over de overige managers. Mocht de leider niet meer beschikbaar zijn om wat voor reden dan ook kan een andere manager zijn taken over nemen.
+
+Om te berekenen hoeveel managers er mogen uitvallen voordat het cluster niet meer kan functioneren wordt de volgende berekening gebruikt: (N-1/2).
```

Comment:
Meestal heeft een formule een uitkomst variable. x = (n-1)/2?
Je haakjes staan ook verkeerd neem ik aan, eerst -1 dan delen door 2.
Waarbij x = het aantal nodes dat mag uitvallen.
Waarbij n = het aantal nodes in je manager group.
Stop de formule ff een een quote of een of andere markdown ding dat die los staat.